src/libmongoc/src/mongoc/mongoc-cmd.c

```diff
@@ -474,6 +474,13 @@ _mongoc_cmd_parts_assemble_mongos (mongoc_cmd_parts_t *parts,
       hedge = mongoc_read_prefs_get_hedge (parts->read_prefs);
    }
 
```

Comment:
I don't think so, but I could be wrong. The guard is around the `_mongoc_cmd_parts_add_read_prefs` function, which is called by `_mongoc_cmd_parts_assemble_mongos` and `mongoc_cmd_parts_assemble`. Since `..._mongod` doesn't call this function, I didn't add any logic there.

Reply:
`_mongoc_cmd_parts_assemble_mongod` uses the `parts->read_prefs->mode` to determine how to set OP_QUERY flags. 

For background, `_mongoc_cmd_parts_assemble_mongod` applies to OP_QUERY. OP_MSG was introduced to supersede OP_QUERY in MongoDB 3.6 (wire version 6).

OP_QUERY is still used on MongoDB 3.6+ servers for monitoring servers with the `hello` / legacy hello commands.

`_mongoc_cmd_parts_assemble_mongod` and `_mongoc_cmd_parts_assemble_mongos` may be able to be removed when the minimum wire version is set to 6 (server version 3.6) in CDRIVER-4098. @eramongodb may be able to confirm or deny that.

If `_mongoc_cmd_parts_assemble_mongod` and `_mongoc_cmd_parts_assemble_mongos` are being removed, I think it is OK to leave them unchanged.

Otherwise, I suggest passing the `mode`  as an argument to `_mongoc_cmd_parts_assemble_mongod`.

Reply:
After CDRIVER-4098 is merged, I think an aggregate will always go through OP_MSG. I do not think changes to `_mongoc_cmd_parts_assemble_mongod` or `_mongoc_cmd_parts_assemble_mongos` are required.

I suggest keeping both `_mongoc_cmd_parts_assemble_mongod` or `_mongoc_cmd_parts_assemble_mongos` consistent, either:
1. Revert the changes to `_mongoc_cmd_parts_assemble_mongos` and and comment noting that an aggregate command can only use OP_MSG.
2. Or change L629 in `_mongoc_cmd_parts_assemble_mongod` from:
```c
         if (parts->read_prefs &&
             parts->read_prefs->mode != MONGOC_READ_PRIMARY) {
            parts->assembled.query_flags |= MONGOC_QUERY_SECONDARY_OK;
         }
```

To:

```c
         if (parts->read_prefs &&
             parts->read_prefs->mode != MONGOC_READ_PRIMARY &&
             !server_stream->must_use_primary) {
            parts->assembled.query_flags |= MONGOC_QUERY_SECONDARY_OK;
         }
```